File: src/login/UserReader.java
package login;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import api.CsvReader;
import api.User;

public class UserReader implements CsvReader {
	private BufferedReader bufferedReader;

	public UserReader(String filename) throws Exception {
		bufferedReader = new BufferedReader(new FileReader(filename));
	}

	@Override
	public List<Object> readData() {
		List<User> users = new ArrayList<User>();
		String currentLine = "";
		String cvsSpliter = ",";

		try {
			while ((currentLine = bufferedReader.readLine()) != null) {
				String[] userValues = currentLine.split(cvsSpliter);
				String[] userBirthdate = userValues[4].split("/");
				
				int year = Integer.parseInt(userBirthdate[2]);
				int month = Integer.parseInt(userBirthdate[1]);
				int day = Integer.parseInt(userBirthdate[0]);
				
				Date birthdate = new Date(year, month, day);
				String login = userValues[0];
				String password = userValues[1];
				String firstname = userValues[2];
				String lastname = userValues[3];
				
				User user = new DefaultUser(login, password, firstname, lastname, birthdate);
				users.add(user);
			}
		} catch (IOException e) {
			e.printStackTrace();
		} finally {
			if (bufferedReader != null) {
				try {
					bufferedReader.close();
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
		}
		return (List<Object>)(List<?>) users;
	}

}

File: src/main/java/universe/user/UserReader.kt
package universe.user

import com.google.common.base.Preconditions.checkArgument
import universe.CsvSeparator.COMMA
import universe.DataReader
import java.io.FileReader
import java.nio.file.Path
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class UserReader(private val usersFilePath: Path, private val birthDateFormater: DateTimeFormatter) : DataReader<User> {

    init {
        checkArgument(usersFilePath.toFile().exists(), "Expected exists users file in path [${usersFilePath.toString()}]");
    }

    override fun readData(): List<User> {
        val users = ArrayList<User>();

        FileReader(usersFilePath.toFile())
                .use { usersReader ->

                    usersReader
                            .readLines()
                            .forEach { line ->
                                val user = readUser(line)
                                users.add(user)
                            }
                }
        return users
    }

    private fun readUser(userRecord: String): User {
        val (login, password, firstname, lastname, birthdate) = userRecord.split(COMMA.separator)
        val userBirthDate = LocalDate.parse(birthdate, birthDateFormater)

        return User(login, password, firstname, lastname, userBirthDate)
    }
}